tests/framework/e2e/etcdctl.go

```diff
@@ -243,3 +243,90 @@ func (ctl *EtcdctlV3) Health() error {
 	return SpawnWithExpects(args, map[string]string{}, lines...)
 
 }
+
+func (ctl *EtcdctlV3) UserAdd(name, password string, opts config.UserAddOptions) (*clientv3.AuthUserAddResponse, error) {
+	args := ctl.cmdArgs()
+	args = append(args, "user", "add")
+	if password == "" {
+		args = append(args, name)
+	} else {
+		args = append(args, fmt.Sprintf("%s:%s", name, password))
+	}
+
+	if opts.NoPassword {
+		args = append(args, "--no-password")
+	}
+
+	args = append(args, "--interactive=false", "-w", "json")
+
+	cmd, err := SpawnCmd(args, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// If no password is provided, and NoPassword isn't set, the CLI will always
+	// wait for a password, send an enter in this case for an "empty" password.
+	if !opts.NoPassword && password == "" {
```

Comment:
Could we simplify code by always providing password in interactive mode?

Reply:
I don't think we need `UserAddOptions` added in [tests/framework/config/client.go](https://github.com/etcd-io/etcd/pull/13819/files#diff-b4d0f6fc9494a0de2abaccf47b3a2e604e4cf751273766c59f82f95dd4c65907). We need to follow the same logic as etcdctl, refer to [user_command.go#L135-L163](https://github.com/etcd-io/etcd/blob/c633e391022ea8064cabb4602fded646a429e851/etcdctl/ctlv3/command/user_command.go#L135-L163).

Reply:
This is the same logic - the CLI will only pass NoPassword: true when --no-password is provided, because it's explicitly distinct behavior from an empty password. (--no-password as documented requires CN auth, but an empty password is also apparently valid on normal accounts?)

Reply:
OK, sounds good. The logic should be OK, and it's a valid case as well. When `password` is empty and `NoPassword` is false, it makes sense to send an enter. It aligns with the existing user experience.